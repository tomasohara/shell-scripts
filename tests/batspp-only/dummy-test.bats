#!/usr/bin/env bats
#
# This test file was generated using Batspp
# https://github.com/LimaBD/batspp
#

# Constants
VERBOSE_DEBUG=""
TEMP_DIR="/tmp/batspp-107056"

# Setup function
# $1 -> test name
function run_setup () {
	test_folder=$(echo $TEMP_DIR/$1-$$)
	mkdir --parents "$test_folder"
	cd "$test_folder" || echo Warning: Unable to "cd $test_folder"
}

# Teardown function
function run_teardown () {
	: # Nothing here...
}

@test "test of line 5" {
	run_setup "test-of-line-5"

	# Assertion of line 5
	shopt -s expand_aliases
	print_debug "$(echo dummy)" "$(echo -e 'dummy\n')"
	[ "$(echo dummy)" == "$(echo -e 'dummy\n')" ]

	run_teardown
}

@test "test of line 9" {
	run_setup "test-of-line-9"

	# Assertion of line 9
	shopt -s expand_aliases
	print_debug "$(date)" "$(echo -e 'Wed Jun 21 06:30:51 CDT 2023\n')"
	[ "$(date)" == "$(echo -e 'Wed Jun 21 06:30:51 CDT 2023\n')" ]

	run_teardown
}

# This prints debug data when an assertion fail
# $1 -> actual value
# $2 -> expected value
function print_debug() {
	echo "=======  actual  ======="
	bash -c "echo \"$1\" $VERBOSE_DEBUG"
	echo "======= expected ======="
	bash -c "echo \"$2\" $VERBOSE_DEBUG"
	echo "========================"
}
