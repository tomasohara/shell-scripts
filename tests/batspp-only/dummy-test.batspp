# Dummy test for debugging purposes
# For use with BatsPP.

# This test will pass
$ echo dummy
dummy

# This test will fail: the time will be different when tested
$ date
Wed Jun 21 06:30:51 CDT 2023
