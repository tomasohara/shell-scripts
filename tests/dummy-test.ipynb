{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "9bddc194",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Dummy test for debugging purposes\n",
    "# For use with BatsPP."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "abb31f9a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dummy\n"
     ]
    }
   ],
   "source": [
    "# This test will pass\n",
    "echo dummy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "ae0d6781",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Wed Jun 21 06:30:51 CDT 2023\n"
     ]
    }
   ],
   "source": [
    "# This test will fail: the time will be different when tested\n",
    "date"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
