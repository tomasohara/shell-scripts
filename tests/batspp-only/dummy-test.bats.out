1..2
ok 1 test-1
not ok 2 test-3
# (in test file batspp-only/dummy-test.bats, line 59)
#   `[ "$(test-3-actual)" == "$(test-3-expected)" ]' failed
# ========== date
#  ==========
# Tue Jul  4 10:25:23 PM +0545 2023
# ========= Wed Jun 21 06:30:51 CDT 2023 =========
# Wed Jun 21 06:30:51 CDT 2023
# ============================
